pkg/p2p/error.go

```diff
@@ -7,14 +7,41 @@ package p2p
 import (
 	"errors"
 	"fmt"
+	"time"
 )
 
-// ErrPeerNotFound should be returned by p2p service methods when the requested
-// peer is not found.
-var ErrPeerNotFound = errors.New("peer not found")
+var (
+	// ErrPeerNotFound should be returned by p2p service methods when the requested
+	// peer is not found.
+	ErrPeerNotFound = errors.New("peer not found")
+	// ErrAlreadyConnected is returned if connect was called for already connected node
+	ErrAlreadyConnected = errors.New("already connected")
+)
+
+// ConnectionBackoffError indicates that connection calls will not be executed until `tryAfter` timetamp.
+// The reason is provided in the wrappped error.
+type ConnectionBackoffError struct {
```

Comment:
definitely not a blocker, but as far as I can see we usually define errors  with naming like `ErrSomething...`, so maybe `ErrConnectionBackoff`? feel free to keep as is though

Reply:
I saw that Custom errors are usually not named this way, only exported error variables. Example for this and the idiom from above: https://blog.golang.org/go1.13-errors